geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -63,27 +63,23 @@ import scala.util.{Failure, Success, Try}
  *  contain multiple features addressed by their featureName.
  */
 class AccumuloDataStore(val connector: Connector,
-                        val authToken: AuthenticationToken,
                         val catalogTable: String,
                         val authorizationsProvider: AuthorizationsProvider,
+                        val auditProvider: AuditProvider,
                         val writeVisibilities: String,
-                        val queryTimeoutConfig: Option[Long] = None,
-                        val queryThreadsConfig: Option[Int] = None,
-                        val recordThreadsConfig: Option[Int] = None,
-                        val writeThreadsConfig: Option[Int] = None,
-                        val cachingConfig: Boolean = false)
+                        val config: AccumuloDataStoreConfig = AccumuloDataStoreConfig())
```

Comment:
We could avoid spreading getOrElse clauses everywhere by dropping the Options for the config. Nice to have defaults with the object/class sometimes like https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/client/BatchWriterConfig.java

``` val config: AccumuloDataStoreConfig = DefaultConfig```

and below in the object

```  val DefaultConfig = AccumuloDataStoreConfig(10, 10, 10, 10, false)``` 

Or make defaults like this:

```
val DefaultConfig = AccumuloDataStoreConfig(
    GeomesaSystemProperties.QueryProperties.QUERY_TIMEOUT_MILLIS.option.map(_.toLong).getOrElse(10),
   GeomesaSystemProperties.QueryProperties.QUERY_RECORD_THREADS.option.map(_.toLong).getOrElse(10),
etc...
```